forgerock-authenticator/src/main/java/org/forgerock/android/auth/DefaultStorageClient.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2020 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import org.forgerock.android.authenticator.Account;
+import org.forgerock.android.authenticator.Mechanism;
+import org.forgerock.android.authenticator.Notification;
+import org.forgerock.android.authenticator.StorageClient;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Data Access Object which implements StorageClient interface and uses SecureSharedPreferences from
+ * forgerock-core SDK to store and load Accounts, Mechanisms and Notifications.
+ */
+public class DefaultStorageClient implements StorageClient {
+
+    //Alias to store keys
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_KEYS = "org.forgerock.android.authenticator.KEYS";
+
+    //Settings to store the data
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_DATA_ACCOUNT = "org.forgerock.android.authenticator.DATA.ACCOUNT";
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_DATA_MECHANISM = "org.forgerock.android.authenticator.DATA.MECHANISM";
+    private static final String ORG_FORGEROCK_SHARED_PREFERENCES_DATA_NOTIFICATIONS = "org.forgerock.android.authenticator.DATA.NOTIFICATIONS";
+
+    //The SharedPreferences to store the data
+    private SharedPreferences accountData;
+    private SharedPreferences mechanismData;
+    private SharedPreferences notificationData;
+
+    private static final String TAG = DefaultStorageClient.class.getSimpleName();
+
+    public DefaultStorageClient(Context context) {
```

Comment:
I suggest using Builder pattern to build the DefaultStorageClient with SharePreferences interface, the developer can provide a different implementation for storage. Android X provides EncryptedSharePreference if they prefer to use that instead of our implementation.

Reply:
It is a good suggestion, but we were not expecting for them to change the implementation of the DefaultStorageClient class. It implements the StorageClient interface, the one that should be used by them to build their own storage system. But I take a not about it.

Reply:
With the SharePreference, you can build automation test with standard SharedPreference but not required to run the test as Instrument test.